### ISSUE TITLE
Pylint disable working incorrectly

### ISSUE THREAD

**Issue by DanJBower** [PyCQA/pylint#7625]
### Bug description

When using `pylint: disable-next` or `pylint: disable` for a comment that violates the rule, it disables the rule for the whole file instead of just the line

*Not working*

    # pylint: disable-next=C0301
    # This line is too long. A line should be less than 100 characters but this is 109 characters and that is bad

    def testMethod():
        print('This line is too long. A line should be less than 100 characters but this is 120 characters and that is bad')

*Not working*

    # This line is too long. A line should be less than 100 characters but this is 109 characters and that is bad # pylint: disable=C0301

    def testMethod():
        print('This line is too long. A line should be less than 100 characters but this is 120 characters and that is bad')

*Working*

    # pylint: disable-next=C0301
    # This line is too long. A line should be less than 100 characters but this is 109 characters and that is bad

    # pylint: enable=C0301
    def testMethod():
        print('This line is too long. A line should be less than 100 characters but this is 120 characters and that is bad')

Whilst above works, I think it is a bug that I have to manually re-enable the rule

### Command used

```shell
pylint Test.py --disable R,C,W,E --enable C0301
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

************* Module Test
Test.py:6:0: C0301: Line too long (120/100) (line-too-long)

-------------------------------------------------------------------
Your code has been rated at 5.00/10

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]
```

### OS / Environment

Windows 10 20H2

**Comment by mbyrnepr2** [comment]
Thanks for this @DanJBower. 

I've tested the first example on 2.15.2 & that one works as expected (message emitted for line 5.). Looks like https://github.com/PyCQA/pylint/issues/7401 could have fixed it.

The second example looks like a bug, however. Modifying the first line to be an assignment with an inlined disable at the end works as expected; so indeed something to do with the comment like you mentioned.

**Comment by DanJBower** [comment]
Thanks for the quick reply @mbyrnepr2

I've just updated to the latest package and the first example is now working for me too. Thanks for pointing that out :)

**Comment by DanielNoord** [comment]
Isn't this a duplicate of https://github.com/PyCQA/pylint/issues/3368?

**Comment by DanJBower** [comment]
@DanielNoord I used C0301 in the example at the start as it was the easy to reproduce. However it seems to apply to any rule when inline disable is used on line that is also a comment

I actually noticed it when I was developing a custom checker and it stopped detecting errors after a specific line. I didn't realise till deleting the disables that that was the issue

I haven't tried the case mentioned in the other issue where disable is placed on a triple quoted string / docstring though so it may have a similar effect?
